### ISSUE TITLE
Validation while typing

### ISSUE THREAD

**Issue by frederic-loui** [jonathanslenders/python-prompt-toolkit#473]
Hi, first of all, thanks for this awesome piece of code.
I checked some part of the code and read documentation but I was just wondering how to enable a prompt validation against a list of element (WordCompleter for example) each time a character (not when carriage return is pressed). (Validation While Typing) I looked at shortcut.py in order to identify if there was an option to the "prompt" function but not luck. Sorry if it is a "silly" question. 
Could you please point me out if there is a knob to do that ?
Thanks !
Frederic

**Comment by jonathanslenders** [comment]
Hi @frederic-loui,

Currently, this is not yet possible. (I never thought about it.) The validation could just like the auto completion and auto suggestion run whenever the input changes.

I take this as a feature request. It's interesting and not too hard to implement.

Thanks,
Jonathan

**Comment by frederic-loui** [comment]
Hi @jonathanslenders,
Wow, I did not expect such a swift answer ! As you mentioned I started to look at auto-completion code in order to trigger validation instead of autocompletion function. But I think you'll integrate this better than me... Hence, many thanks for considering this as an enhancement. As complement of information, this behaviour is typical from CLI used to run TELECOM hardware equipment like Juniper JUNOS or Cisco IOS. One of the most powerful and flexible CLI being JUNOS CLI.  Thanks again for your support for all of your hard work.
Cheers, Frederic
